### PR TITLE
feat(dal): Adds a Module Node to the graph for each of the installed builtins

### PR DESCRIPTION
**Pull request by stack72**
This isn't connected to anything at this time. I have also not removed the InstalledPkgs impl OR the storing in the old database structure

### CODE REVIEW

**Comment by stack72** [comment]
bors try

File: lib/dal/src/layer_db_types/content_types.rs [hunk 1]
```diff
@@ -258,6 +260,18 @@ pub struct InputSocketContentV1 {
     pub connection_annotations: Vec<ConnectionAnnotation>,
 }
 
+#[derive(Debug, Clone, EnumDiscriminants, Serialize, Deserialize, PartialEq)]
+pub enum ModuleContent {
+    V1(ModuleContentV1),
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub struct ModuleContentV1 {
```
Comment (britmyerss) [inline lib/dal/src/layer_db_types/content_types.rs:269]: do we also need: 
`display_name` `name` etc?

Comment (stack72) [reply lib/dal/src/layer_db_types/content_types.rs:269]: That's not part of the module - that's part of the Schema / Schema Variant. The module just has a name and a root_hash.

Comment (britmyerss) [reply lib/dal/src/layer_db_types/content_types.rs:269]: `export interface LocalModuleSummary {
  name: string;
  hash: ModuleHash;
  isBuiltin: boolean;
}

export interface LocalModuleDetails {
  name: string;
  version: string;
  description: string;
  createdAt: IsoDateString;
  createdBy: string;
  schemas: string[];
  funcs: PkgFuncView[];
  hash: ModuleHash;
  kind: "module" | "workspaceExport";
}`
I think there's a name/description/etc. for that too - here's from app/web

Comment (nickgerace) [reply lib/dal/src/layer_db_types/content_types.rs:269]: I wonder if modules implicitly only have one schema (not saying how it should always be, just how it is now) which is influencing what's returned in that route?

Comment (britmyerss) [reply lib/dal/src/layer_db_types/content_types.rs:269]: I don't think so.. even now we have multiple schemas in a module

**Comment by stack72** [comment]
bors try

**Comment by stack72** [comment]
bors merge

**Pull request merged**